kong/client.go

```diff
@@ -264,17 +265,36 @@ func (c *Client) Status(ctx context.Context) (*Status, error) {
 }
 
 // Root returns the response of GET request on root of
-// Admin API (GET /).
+// Admin API (GET / by default or GET /kong when the client has workspace defined)
 func (c *Client) Root(ctx context.Context) (map[string]interface{}, error) {
-	req, err := c.NewRequest("GET", "/", nil, nil)
+	result, err := c.getMap(ctx, "/kong")
+	if err != nil {
+		if IsNotFoundErr(err) {
+			return c.getMap(ctx, "/")
+		}
+		return nil, err
+	}
+	return result, nil
+}
+
+func (c *Client) getMap(ctx context.Context, queryPath string) (map[string]interface{}, error) {
+	req, err := c.NewRequest("GET", queryPath, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+	_, err = c.Do(ctx, req, &result)
 	if err != nil {
 		return nil, err
 	}
+	return result, nil
+}
 
-	var root map[string]interface{}
-	_, err = c.Do(ctx, req, &root)
+func (c *Client) Kong(ctx context.Context) (*Kong, error) {
+	root, err := c.Root(ctx)
 	if err != nil {
 		return nil, err
 	}
-	return root, nil
+	return getKong(root)
```

Comment:
I don't think this pattern is good.
We should instead give a utility function to the user which takes the output `/` or `/kong` and answers the questions around support for tags, enterprise, etc.
The `kong` packages is intended for talking to the API. Interpreting these specifics should not be part of this package.